Saule/Http/JsonApiProcessor.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Web.Http.Filters;
+using Saule.Serialization;
+
+namespace Saule.Http
+{
+    /// <summary>
+    /// Processes JSON API responses
+    /// </summary>
+    public class JsonApiProcessor
```

Comment:
Do you think we can call this from the delegating handler? Now the code is duplicated. I think since this is public, it would be awesome if the delegating handler used the same code-path.